OpenRA.Mods.Common/LoadScreens/BlankLoadScreen.cs

```diff
@@ -61,24 +61,21 @@ public void StartGame(Arguments args)
 				return;
 			}
 
-			// Join a server directly
 			var connect = string.Empty;
-			if (args != null)
+
+			if (!string.IsNullOrEmpty(Game.Launch.Connect))
 			{
-				if (args.Contains("Launch.Connect"))
-					connect = args.GetValue("Launch.Connect", null);
+				connect = Game.Launch.Connect;
+				Game.Launch.Connect = null;
+			}
 
-				if (args.Contains("Launch.URI"))
-				{
-					connect = args.GetValue("Launch.URI", null);
-					if (connect != null)
-					{
-						connect = connect.Replace("openra://", "");
-						connect = connect.TrimEnd('/');
-					}
-				}
+			if (!string.IsNullOrEmpty(Game.Launch.URI))
+			{
+				connect = Game.Launch.URI.Replace("openra://", "").TrimEnd('/');
+				Game.Launch.URI = null;
```

Comment:
Don't persist it.  Pass the object to the things that want it, then throw it away when you're done.